tensorflow_datasets/testing/dataset_builder_testing.py

```diff
@@ -359,6 +359,9 @@ def _bytes_flatten(element):
         element = element.decode("latin-1")
       element = element.encode("utf-8")
       ret += element
+    elif isinstance(element, (tf.RaggedTensor,
+                              tf.compat.v1.ragged.RaggedTensorValue)):
+      ret += str(element).encode("utf-8")
```

Comment:
Could you add a TODO:
```
# TODO(tfds): Checksum should be computed from the `ragged.to_list()` value.
```

or try with
```
try with `str(element.to_list())
```
instead ?

Reply:
Sure

On a separate note: I noticed that `_bytes_flatten()` returns a different value every time I call it with the same string tensor (e.g., `tf.constant([b'foo', b'bar'])`). Even calling `numpy()` on the tensor doesn't help.

Reply:
`_bytes_flatten` shouldn't be called directly on tensor, because nothing guarantee that the __str__ representation is constant.
This is why it should first be converted to values which always produce deterministic output.

`ret += tensor.numpy().tobytes()` or `ret += _bytes_flatten(tensor.numpy())` should works.

Reply:
That's what I thought, too, but I think there must be a bug in NumPy, because it is not the case. Try this:

```python
import numpy as np
import tensorflow as tf

for _ in range(5):
    obj = np.array([b"foo", b"bar"])
    print(bytes(obj))
print()

for _ in range(5):
    obj = tf.constant([b"foo", b"bar"])
    print(bytes(obj))
print()

for _ in range(5):
    obj = tf.constant([b"foo", b"bar"]).numpy()
    print(bytes(obj))
print()

for _ in range(5):
    obj = tf.constant([b"foo", b"bar"]).numpy()
    print(obj.tobytes())
print()
```

Reply:
Interesting. Thanks for catching this. It's good to know.
I think the issue is that `tf.constant([b"foo", b"bar"]).numpy().dtype == object` while `np.array([b"foo", b"bar"]).dtype = 'S3'`.
Opening a bug, even if it's probably not a huge priority.